src/test/java/com/diffmin/AppTest.java

```diff
@@ -216,6 +238,34 @@ private static void runTests(TestResources sources) throws Exception {
                             .getOrCreate(retrievedFirstType);
             String patchedProgram = SpoonUtil.displayModifiedModel(patchedCtModel);
             assertEquals(cu.prettyprint(), patchedProgram, "Prev file was not patched correctly");
+
+            // check the root origination of each element
+            List<String> insertedLines = Files.readAllLines(sources.insertedPaths);
+
+            Iterator<CtElement> it = patchedCtModel.getRootPackage().descendantIterator();
+
+            while (it.hasNext()) {
+                CtElement element = it.next();
+                if (isToIgnore(element, insertedLines)) {
+                    continue;
+                }
```

Comment:
You can most likely replace this like so:

```suggestion
                if (!element.getPosition().isValidPosition()) {
                    continue;
                }
```
```